app/actions/local/channel.ts

```diff
@@ -237,32 +239,51 @@ export async function resetMessageCount(serverUrl: string, channelId: string) {
     }
 }
 
-export async function storeMyChannelsForTeam(serverUrl: string, teamId: string, channels: Channel[], memberships: ChannelMembership[], prepareRecordsOnly = false, isCRTEnabled?: boolean) {
+const resolveAndflattenModels = async (operator: ServerDataOperator, modelPromises: Array<Promise<Model[]>>, description: string, prepareRecordsOnly: boolean) => {
+    const models = await Promise.all(modelPromises);
+    if (!models.length) {
+        return {models: []};
+    }
+
+    const flattenedModels = models.flat();
+
+    if (prepareRecordsOnly) {
+        return {models: flattenedModels};
+    }
+
+    if (flattenedModels.length) {
+        await operator.batchRecords(flattenedModels, description);
+    }
+
+    return {models: flattenedModels, error: undefined};
+};
+
+export async function storeAllMyChannels(serverUrl: string, channels: Channel[], memberships: ChannelMembership[], isCRTEnabled: boolean, prepareRecordsOnly = false) {
     try {
         const {operator} = DatabaseManager.getServerDatabaseAndOperator(serverUrl);
+        const cs = new Set(channels.map((c) => c.id));
         const modelPromises: Array<Promise<Model[]>> = [
-            ...await prepareMyChannelsForTeam(operator, teamId, channels, memberships, isCRTEnabled),
+            ...await prepareAllMyChannels(operator, channels, memberships.filter((m) => cs.has(m.channel_id)), isCRTEnabled),
         ];
 
-        const models = await Promise.all(modelPromises);
-        if (!models.length) {
-            return {models: []};
-        }
-
-        const flattenedModels = models.flat();
-
-        if (prepareRecordsOnly) {
-            return {models: flattenedModels};
-        }
+        return resolveAndflattenModels(operator, modelPromises, 'storeAllMyChannels', prepareRecordsOnly);
```

Comment:
I always struggle a bit with how JavaScript handles this.

At this point, we are returning the promise, and therefore, exiting the `try/catch` block. So... if a exception occur inside the promise, the exception will not be handled here.

So... in this particular case, the catch will only catch the first promise resolution of `prepareAllMyChannels`, but not the resolution of `resolveAndFlattenModels` or the resolution of `prepareChannels` inside  `prepareAllMyChannels`.

I think the linter understands you are inside a try/catch and allows you to do this:
```suggestion
        return await resolveAndflattenModels(operator, modelPromises, 'storeAllMyChannels', prepareRecordsOnly);
```

Reply:
Sure, but would it get caught by the caller?  In this case /remote/channel.ts has a try-catch block there too.

Reply:
I've confirmed that the await is not needed due to try..catch on the caller catching this.

Reply:
The point being is, there's try..catch on the callers.  So errors within the promise will get bubbled up and caught by the callers.

Reply:
By looking at the function, there is nothing that shows that it is going to throw. On the contrary, having the whole body of the function inside a try catch block should be an indication that you don't expect this function to throw.

If the callers of this function have try catch, great, it won't fail. But that doesn't mean this is not wrong. Not only that, we will lose part of the context of what is failing, since the catch will be higher up, instead of here, where it will clearly log that the the problem is specifically in `storeAllMyChannels`.

Reply:
I made a new commit to add a try..catch in resolveAndFlattenModels.  So if any of the models fail for whatever reason, we'll get a more isolated error output.